Sources/Plasma/PubUtilLib/plVault/plVaultClientApi.cpp

```diff
@@ -728,13 +696,13 @@ static void SaveDirtyNodes () {
     if (!s_nextSaveMs || signed(s_nextSaveMs - currTimeMs) <= 0) {
         s_nextSaveMs = (currTimeMs + kSaveUpdateIntervalMs) | 1;
         unsigned bytesWritten = 0;
-        for (RelVaultNodeLink * link = s_nodes.Head(); link; link = s_nodes.Next(link)) {
+        for (const auto& [nodeId, node] : s_nodes) {
```

Comment:
```suggestion
        for (auto [nodeId, node] : s_nodes) {
```

Correct me if I'm wrong, but I think that C++17 structured bindings for tuples and pairs will return references.

Reply:
That's a good question... When you dereference an iterator, that definitely results in a reference, but I'm not sure that the reference-ness is "transferred" to a variable declared with `auto`. (The structured binding might be irrelevant here - I think it would be the same thing if you store the entire pair in an `auto` variable, but I'm not sure.)

To be honest, I'm a bit out of my depth with this. I'm mostly going by the [examples on cppreference.com](https://en.cppreference.com/w/cpp/container/unordered_map#Example), which consistently use `const auto& ...` and not just `auto ...` when doing a range `for` over a map.

Reply:
Maybe @zrax can clarify this for us... I noticed that cppreference was using `const auto&` for maps *after* I submitted the review, but I'm not sure why exactly.

Reply:
From what I understand, the structured binding just extracts the declarations as if you had declared them individually in separate declarations.  Therefore, `auto [k, v] = tuple` will copy (or move) the values from `tuple` into non-reference types for `k` and `v`.  I think where it gets fuzzy/interesting is that the `std::unordered_map` iterator returns ***references to `std::pair<K, V>` objects***, and not a pair of references (!!).  So although a classic iterator would still give you references to the data, extracting them with a structured binding will give you copies/moves unless you use a reference in your binding.

Playground: https://godbolt.org/z/4xsPYG6vs

TLDR:  `const auto&` appears to be correct here.